src/libs/ajax/data-table-providers/WDSDataTableProvider.ts

```diff
@@ -0,0 +1,117 @@
+import _ from 'lodash/fp'
+import { Ajax } from 'src/libs/ajax'
+import { DataTableFeatures, DataTableProvider, DeleteTableFn, DownloadTsvFn, EntityMetadata, EntityQueryOptions, EntityQueryResponse, GetMetadataFn, GetPageFn } from 'src/libs/ajax/data-table-providers/DataTableProvider'
+
+// interface definitions for WDS payload responses
+interface AttributeSchema {
+  name: string,
+  datatype: string,
+  relatesTo?: string
+}
+
+export interface RecordTypeSchema {
+  name: string,
+  count: number,
+  attributes: AttributeSchema[]
+}
+
+interface SearchRequest {
+  offset: number,
+  limit: number,
+  sort: 'asc' | 'ASC' | 'desc' | 'DESC',
+  sortAttribute: string
+}
+
+interface RecordResponse {
+  id: string,
+  type: string,
+  attributes: { [index: string]: any } // truly "any" here; the backend Java representation is Map<String, Object>
+}
+
+export interface RecordQueryResponse {
+  searchRequest: SearchRequest,
+  totalRecords: number,
+  records: RecordResponse[]
+}
+
+
+export class WDSDataTableProvider implements DataTableProvider {
+  constructor(workspaceId: string) {
+    this.workspaceId = workspaceId
+  }
+
+  workspaceId: string
+
+  features: DataTableFeatures = {
+    supportsTsvDownload: false,
+    supportsTsvAjaxDownload: true,
+    supportsTypeDeletion: true,
+    supportsTypeRenaming: false,
+    supportsExport: false,
+    supportsPointCorrection: false,
+    supportsFiltering: false
+  }
+
+  transformPage: (arg0: RecordQueryResponse, arg1: string, arg2: EntityQueryOptions) => EntityQueryResponse = (wdsPage: RecordQueryResponse, recordType: string, queryOptions: EntityQueryOptions) => {
```

Comment:
Don't need to specify types twice here. Similar comments for other methods in this class.

```suggestion
  transformPage = (wdsPage: RecordQueryResponse, recordType: string, queryOptions: EntityQueryOptions): EntityQueryResponse => {
```